metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/ConnectorFactory.java

```diff
@@ -57,11 +57,18 @@ default ConnectorPartitionService getPartitionService() {
     }
 
     /**
-     * Returns the name of the connector.
+     * Returns the name of the catalog.
      *
-     * @return Returns the name of the connector.
+     * @return Returns the name of the catalog.
      */
-    String getName();
+    String getCatalogName();
+
+    /**
+     * Returns the name of the catalog shard.
+     *
+     * @return Returns the name of the catalog shard.
+     */
+    String getCatalogShardName();
```

Comment:
Should this return an `Optional` to reflect that it could be null or empty?

Reply:
This will be the same as catalogName when there is only one shard for one catalog.